projects/cloud/app/services/project_fetch_service.rb

```diff
@@ -1,16 +1,23 @@
 # frozen_string_literal: true
 
 class ProjectFetchService < ApplicationService
-  attr_reader :name, :account_name
+  module Error
+    Unauthorized = Class.new(StandardError)
+  end
+
+  attr_reader :name, :account_name, :current_user
 
-  def initialize(name:, account_name:)
+  def initialize(name:, account_name:, current_user:)
     super()
     @name = name
     @account_name = account_name
+    @current_user = current_user
   end
 
   def call
     account = Account.find_by(name: account_name)
-    Project.find_by(account_id: account.id, name: name)
+    project = Project.find_by(account_id: account.id, name: name)
+    raise Error::Unauthorized unless ProjectPolicy.new(current_user, project).show?
```

Comment:
Make sure you rescue from this one at the mutation level and return a standard response agreed with the frontend.

Reply:
good point, done!